### PR TITLE
ANO generic interface

### PR DESCRIPTION
**Pull request by cliffburdick**
Add a pluggable interface to the ANO so the backend can be chosen automatically based on what is appropriate from the config file

### CODE REVIEW

File: operators/advanced_network/CMakeLists.txt [hunk 1]
```diff
@@ -50,8 +47,8 @@ set(DPDK_EXTRA_LIBS -Wl,--no-whole-archive -lmlx5 -libverbs -pthread -lnuma -ldl
 
 pkg_check_modules(DPDK REQUIRED libdpdk)
 
-target_include_directories(advanced_network_rx INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
-target_include_directories(advanced_network_tx INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+# target_include_directories(advanced_network_rx INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (wendell-hom) [inline operators/advanced_network/CMakeLists.txt:50]: @cliffburdick is it still useful to leave these here?

**Comment by wendell-hom** [comment]
@cliffburdick I trust these changes have all been tested?  I don't think we have any existing automated testing for ANO from holohub

**Comment by cliffburdick** [comment]
> @cliffburdick I trust these changes have all been tested? I don't think we have any existing automated testing for ANO from holohub

Yes, they've been tested on my end. We have no hardware in CICD to test it automatically

File: operators/advanced_network/CMakeLists.txt (hunk 1 above)
Comment (cliffburdick) [reply operators/advanced_network/CMakeLists.txt]: Fixed

**Pull request merged**